iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -1013,16 +1013,22 @@ def transform_index_backwards(
             iters = self.mapping.iters
             mapping = self.mapping.dynamic_val_mappings[i]
             subs = {v: k for k, v in zip(iters, mapping.keys())}
-            return {k: v.apply_expr(subs[k], mapping[k]) for k, v in index.items()}
+            return {
+                k: v.apply_expr(subs[k], mapping[k])
+                for k, v in zip(arg.type.symbolic_shape, index.values())
```

Comment:
Why not use index.items() instead of splitting this into a type.symbolic_shape and index.values()?